Engine/Src/Ancona/Core2D/Systems/PathFollowerSystem.cpp

```diff
@@ -0,0 +1,107 @@
+#include "PathFollowerSystem.hpp"
+
+using namespace ild;
+
+/* Component */
+PathFollowerComponent::PathFollowerComponent() {
+}
+
+void PathFollowerComponent::Update(float delta) {
+    if (_pathComponent->vertices().size() == 0) {
+        return;
+    }
+        
+    if (_action->Done()) {
+        StartNextPathSegment();
+    }
+}
+
+void PathFollowerComponent::StartNextPathSegment() {
+    auto lastVertexIndex = _nextVertexIndex;
+    _nextVertexIndex = NextVertexIndex();
```

Comment:
Hmmmm. NextVertexIndex doesn't really indicate that it would change _followDirection, not sure what we could rename it though